src/Microsoft.AspNet.Server.Kestrel/Http/SocketOutput.cs

```diff
@@ -48,6 +59,9 @@ public class SocketOutput : ISocketOutput
             _connectionId = connectionId;
             _log = log;
             _tasksPending = new Queue<TaskCompletionSource<object>>();
+
+            _head = memory.Lease();
```

Comment:
I almost wrote this to do lazy leasing, but decided against it after I figured that we should almost never be creating a new SocketOutput without writing to it. What I hadn't really taken into account is the potential negative perf impact of leasing blocks longer than necessary.

Currently `null` means the socket has been closed, but I guess I can add a bool to track that instead.